# Dataset di partenza
(https://www.harmo.org/jsirwin/RoundHillDiscussion.html)

In [1]:
from util import load_from_excel

fname1 = 'RHILL1Update01.xls'
fname2 = 'RHILL2Update01.xls'

data1 = load_from_excel(fname1, sheet_name=2)
data2 = load_from_excel(fname2, sheet_name=2)

meteo1 = load_from_excel(fname1, sheet_name=1)
meteo2 = load_from_excel(fname2, sheet_name=1)

correction1 = load_from_excel(fname1, sheet_name=0)
correction2 = load_from_excel(fname2, sheet_name=0)

# Dati di misura
I dati di misura sono stati ricavati da due esperimenti realizzati rispettivamente nelle date:
- Agosto 1954 - Novembre 1955
- Settembre 1957 - Dicembre 1957
Entrambi gli esperimenti sono stati effettuati nella stessa zona (Round Hill Field Station, Massachusetts Institute of Technology) con un setup simile.
Dettagli:
## Round Hill 1
### Emissione del gas
- Fonte di SO2 situata ad un'altezza di 30 cm dal livello del terreno
### Misure di concentrazione
- Le misure sono state effettuate in 3 semicirconferenze di raggio 50, 100, 200 m, con centro nella fonte del gas, tramite sensori posti lungo la circonferenza ad una distanza di 1.5 gradi l'uno dall'altro. Ogni sensore misura la concentrazione media di SO2 nell'aria per un periodo di 10 minuti

## Round Hill 2
### Emissione del gas
- Fonte di SO2 situata ad un'altezza di 1.5 m dal livello del terreno
### Misure di concentrazione
- Le misure sono state effettuate ad un'altezza di 1.5 m in 3 semicirconferenze di raggio 50, 100, 200 m con centro nella fonte del gas. In questo caso i sensori 1-6 e 106-111 sono distanziati di 3 gradi, mentre gli altri di 1.5 gradi. Le misure di concentrazione media sono state effettuate per un periodo di 10, 3, 1.5 min (Tutti gli intervalli di misurazione iniziano nello stesso istante)

### Misure meteorologiche (Round Hill 1 e Round Hill 2)
Durante l'esperimento sono state effettuate le seguenti misurazioni, in un periodo di 20 minuti centrato nei 10 minuti di misura:
- Velocità del vento media
- Temperatura
Questi due parametri sono stati misurati a diverse altezze (1.5, 3, 6 e 12 m)
- Distribuzione della direzione del vento
Quest'ultima è stata misurata ad un'altezza di 2 m

## Correzioni
I due dataset presentano dei fattori di correzione da applicare alle misure di concentrazione. Questi fattori moltiplicativi tengono conto dell'evaporazione della fonte di SO2, avvenuta più velocemente del previsto a causa delle temperature, portando quindi a misure di concentrazione più elevate


In [2]:
data1.head(10)

Unnamed: 0                                         Unnamed: 1 Unnamed: 2  \
0        NaN  Summary Excel Worksheet for Round Hill I (Disp...        NaN   
1        NaN                                                NaN        NaN   
2        NaN  Runs for which tracer may have flowed beneath ...        NaN   
3         50                                                  6          7   
4        100                                                NaN        NaN   
5        200                                                NaN        NaN   
6        NaN                                                NaN        NaN   
7        NaN  From my analysis of Fig. 1, Post #1 looks to b...        NaN   
8        NaN     If you change D11, you alter all Post Azimuths        NaN   
9        NaN                                                NaN        NaN   

  Unnamed: 3 Unnamed: 4  Unnamed: 5  Unnamed: 6  
0        NaN        NaN         NaN         NaN  
1        NaN        NaN         NaN         NaN  
2        NaN        NaN         NaN         NaN  
3          9         10        13.0         NaN  
4          9         10        13.0         NaN  
5          9         10         NaN         NaN  
6        NaN        NaN         NaN         NaN  
7        NaN        NaN         NaN         NaN  
8        NaN        NaN         NaN         NaN  
9         25        NaN         NaN         NaN

In [3]:
data2.head(10)

Unnamed: 0                                         Unnamed: 1 Unnamed: 2  \
0        NaN  Summary Excel Worksheet for Round Hill II (Dis...        NaN   
1        NaN                                                NaN        NaN   
2        NaN  From my analysis of Fig. 30, Post #1 looks to ...        NaN   
3        NaN      If you change D6, you alter all Post Azimuths        NaN   
4        NaN                                                NaN        NaN   
5        NaN                                                NaN        NaN   
6        NaN                                                NaN        NaN   
7        NaN  Appendix B. Table I. Sulfur-dioxide concentrat...        NaN   
8        NaN  and 10-min frequency distributions of azimuth ...        NaN   
9        NaN                                                NaN        NaN   

  Unnamed: 3 Unnamed: 4 Unnamed: 5  Unnamed: 6  \
0        NaN        NaN        NaN         NaN   
1        NaN        NaN        NaN         NaN   
2        NaN        NaN        NaN         NaN   
3        NaN        NaN        NaN         NaN   
4         15        NaN        NaN         NaN   
5        NaN        NaN        NaN         NaN   
6        NaN        NaN        NaN         NaN   
7        NaN        NaN        NaN         NaN   
8        NaN        NaN        NaN         NaN   
9        NaN        NaN        NaN         NaN   

                          Unnamed: 7  Unnamed: 8 Unnamed: 9  ... Unnamed: 12  \
0                                NaN         NaN        NaN  ...         NaN   
1                                NaN         NaN        NaN  ...         NaN   
2                                NaN         NaN        NaN  ...         NaN   
3                                NaN         NaN        NaN  ...         NaN   
4                                NaN         NaN        NaN  ...         NaN   
5                                NaN         NaN        NaN  ...         NaN   
6                                NaN         NaN        NaN  ...         NaN   
7                                NaN         NaN        NaN  ...         NaN   
8                                NaN         NaN        NaN  ...         NaN   
9  Horizontal Sampling Array at 1.5m         NaN        NaN  ...         NaN   

  Unnamed: 13  Unnamed: 14 Unnamed: 15  Unnamed: 16  Unnamed: 17  \
0         NaN          NaN         NaN          NaN          NaN   
1         NaN          NaN         NaN          NaN          NaN   
2         NaN          NaN         NaN          NaN          NaN   
3         NaN          NaN         NaN          NaN          NaN   
4         NaN          NaN         NaN          NaN          NaN   
5         NaN          NaN         NaN          NaN          NaN   
6         NaN          NaN         NaN          NaN          NaN   
7         NaN          NaN         NaN          NaN          NaN   
8         NaN          NaN         NaN          NaN          NaN   
9         NaN          NaN         NaN          NaN          NaN   

                 Unnamed: 18 Unnamed: 19 Unnamed: 20 Unnamed: 21  
0                        NaN         NaN         NaN         NaN  
1                        NaN         NaN         NaN         NaN  
2                        NaN         NaN         NaN         NaN  
3                        NaN         NaN         NaN         NaN  
4                        NaN         NaN         NaN         NaN  
5                        NaN         NaN         NaN         NaN  
6                        NaN         NaN         NaN         NaN  
7                        NaN         NaN         NaN         NaN  
8                        NaN         NaN         NaN         NaN  
9  VERTICAL SAMPLING RESULTS         NaN         NaN         NaN  

[10 rows x 22 columns]

In [7]:
meteo1.head(10)

Unnamed: 0 Unnamed: 1                                         Unnamed: 2  \
0          NaN        NaN                                                NaN   
1          NaN        NaN                               Missing values = -99   
2          NaN        NaN                                On-Site Meteorology   
3          NaN        NaN  All values reported are for the 10-min gas-sam...   
4          NaN        NaN                                                NaN   
5          NaN        NaN                          U - horizontal wind speed   
6          NaN        NaN  Se - standard deviation of the bivane elevatio...   
7          NaN        NaN  Sa - standard deviation of the bivane horizont...   
8          NaN        NaN                                                NaN   
9          NaN        NaN  Numbers following indicate height of the meteo...   
10         NaN        NaN                                                NaN   
11         NaN        NaN  Appendix A. Table III. Summary of meteorologic...   
12         NaN        NaN                                                NaN   
13         NaN        NaN                                          At Source   
14         NaN        NaN             --------------------------------------   
15         NaN        Run                                                 U2   
16         NaN     Number                                              (m/s)   
17         NaN          1                                               2.17   
18         NaN          2                                               4.04   
19         NaN          3                                               2.84   
20         NaN          4                                               2.62   
21         NaN          5                                               5.25   
22         NaN          6                                               4.21   
23         NaN          7                                               8.76   
24         NaN          8                                               3.86   
25         NaN          9                                               7.46   
26         NaN         10                                               7.12   
27         NaN         11                                               7.76   
28         NaN         12                                                4.3   
29         NaN         13                                                9.2   
30         NaN         14                                               4.38   
31         NaN         15                                               4.46   
32         NaN         16                                               4.56   
33         NaN         17                                                2.4   
34         NaN         18                                               2.94   
35         NaN         19                                               2.71   
36         NaN         20                                               1.87   
37         NaN         21                                               1.49   
38         NaN         22                                               1.79   
39         NaN         23                                               1.74   
40         NaN         24                                               1.98   
41         NaN         25                                               1.63   
42         NaN         26                                               2.79   
43         NaN         27                                               3.32   
44         NaN         28                                               1.79   
45         NaN         29                                               1.78   

   Unnamed: 3 Unnamed: 4                                         Unnamed: 5  \
0         NaN        NaN                                                NaN   
1         NaN        NaN                                                NaN   
2     

In [8]:
meteo2.head(10)

Unnamed: 0 Unnamed: 1                                         Unnamed: 2  \
0          NaN        NaN                                                NaN   
1          NaN        NaN                               Missing values = -99   
2          NaN        NaN                                On-Site Meteorology   
3          NaN        NaN  All values reported are for the 10-min gas-sam...   
4          NaN        NaN                                                NaN   
5          NaN        NaN                          U - horizontal wind speed   
6          NaN        NaN  Se - standard deviation of the bivane elevatio...   
7          NaN        NaN  Sa - standard deviation of the bivane horizont...   
8          NaN        NaN                                                NaN   
9          NaN        NaN  Numbers following indicate height of the meteo...   
10         NaN        NaN                                                NaN   
11         NaN        NaN  Appendix B. Table IV. Summary of meteorologica...   
12         NaN        NaN                                                NaN   
13         NaN        NaN                                                NaN   
14         NaN        NaN                                        Cup Anemom.   
15         NaN        NaN                                    At Source (2 m)   
16         NaN     RHill2                          -------------------------   
17         NaN        Run                                             U(m/s)   
18         NaN          1                                               2.14   
19         NaN          2                                               5.15   
20         NaN          3                                                  3   
21         NaN          4                                               3.98   
22         NaN          5                                               1.65   
23         NaN          6                                               2.61   
24         NaN          7                                               3.86   
25         NaN          8                                               3.31   
26         NaN          9                                               4.39   
27         NaN         10                                               4.07   

   Unnamed: 3                                         Unnamed: 4 Unnamed: 5  \
0         NaN                                                NaN        NaN   
1         NaN                                                NaN        NaN   
2         NaN                                                NaN        NaN   
3         NaN                                                NaN        NaN   
4         NaN                                                NaN        NaN   
5         NaN                                                NaN        NaN   
6         NaN                                                NaN        NaN   
7         NaN                                                NaN        NaN   
8         NaN                                                NaN        NaN   
9         NaN                                                NaN        NaN   
10        NaN                                                NaN        NaN   
11        NaN                                                NaN        NaN   
12        NaN                                                NaN        NaN   
13        NaN                                                NaN        NaN   
14        NaN                                                NaN        NaN   
15        NaN                        Horizontal Wind Speed (m/s)        NaN   
16        NaN    --------------------------------------------...        NaN   
17    Sa(deg)                                               U1.5       U3.0   
18        8.4                                               2.52       2.87   
19       16.5                                               5.36       6.12   
20       12.9           

In [9]:
correction1.head(10)

Unnamed: 0                                         Unnamed: 1 Unnamed: 2  \
0          NaN  Appendix A. Table II. Concentrations listed in...        NaN   
1          NaN  be multiplied by values listed here, to compen...        NaN   
2          NaN  evaporational loss of impinger solution during...        NaN   
3          NaN                                                NaN        NaN   
4          NaN                                                NaN        NaN   
5          NaN                                                Run          Q   
6          NaN                                             Number      (g/s)   
7          NaN                                                  1          1   
8          NaN                                                  2       4.43   
9          NaN                                                  3      11.49   
10         NaN                                                  4       6.67   
11         NaN                                                  5       8.82   
12         NaN                                                  6       7.24   
13         NaN                                                  7       9.16   
14         NaN                                                  8       9.31   
15         NaN                                                  9       9.84   
16         NaN                                                 10      10.04   
17         NaN                                                 11       9.96   
18         NaN                                                 12       9.92   
19         NaN                                                 13       9.98   
20         NaN                                                 14       9.96   
21         NaN                                                 15       9.95   
22         NaN                                                 16       9.96   
23         NaN                                                 17       9.88   
24         NaN                                                 18       8.12   
25         NaN                                                 19       8.58   
26         NaN                                                 20       8.48   
27         NaN                                                 21       6.23   
28         NaN                                                 22       6.42   
29         NaN                                                 23        6.6   
30         NaN                                                 24       5.72   
31         NaN                                                 25       5.77   
32         NaN                                                 26       6.87   
33         NaN                                                 27       6.65   
34         NaN                                                 28       7.94   
35         NaN                                                 29       7.96   

                 Unnamed: 3  Unnamed: 4  Unnamed: 5  
0                       NaN         NaN         NaN  
1                       NaN         NaN         NaN  
2                       NaN         NaN         NaN  
3                       NaN         NaN         NaN  
4                       NaN         NaN         NaN  
5   Evaporation Corrections         NaN         NaN  
6                        50      100.00      200.00  
7                      0.93        0.93        0.92  
8                      0.91        0.94        0.93  
9                       0.9        0.91        0.92  
10                     0.91        0.87        0.86  
11                     0.95        0.94        0.96  
12                     0.96        0.91        0.92  
13                     0.95        0.94        0.96  
14                     0.96        0.94        0.93  
15                     0.98        0.99        0.97  
16                     0.96        0.97        0.97  
17                     0.98        0.97        0.98  
18               

In [10]:
correction2.head(10)

Unnamed: 0                                         Unnamed: 1  \
0          NaN                                                NaN   
1          NaN  Appendix B. Table II. Summary of source streng...   
2          NaN                                                NaN   
3          NaN                                                NaN   
4          NaN                                             RHill2   
5          NaN                                                Run   
6          NaN                                                  1   
7          NaN                                                  2   
8          NaN                                                  3   
9          NaN                                                  4   
10         NaN                                                  5   
11         NaN                                                  6   
12         NaN                                                  7   
13         NaN                                                  8   
14         NaN                                                  9   
15         NaN                                                 10   
16         NaN                                                NaN   
17         NaN  Note #1. During the last 8 minutes of release ...   
18         NaN  strength decreased from about 70 g/s to about ...   
19         NaN             should only affect the 10-min results.   
20         NaN                                                NaN   
21         NaN  Note #2. For release period #1, for the 0.5-mi...   
22         NaN                                                NaN   
23         NaN  Concentration values listed in Appendix B. Tab...   
24         NaN  for evaporational loss of impinger solution du...   

                    Unnamed: 2 Unnamed: 3 Unnamed: 4          Unnamed: 5  \
0                          NaN        NaN        NaN                 NaN   
1                          NaN        NaN        NaN                 NaN   
2                          NaN        NaN        NaN                 NaN   
3                          NaN        NaN        NaN                 NaN   
4        Source Strength (g/s)        NaN        NaN         Corrections   
5                       10-min      3-min    0.5-min              10-min   
6                         45.1       45.1       45.1                 0.9   
7                          109        114        115                0.91   
8                         53.4       53.4       53.4                0.93   
9                          100        108        108                0.92   
10                        49.2       49.2       49.2                0.97   
11                        53.1       54.2       54.2                0.96   
12                        56.5       56.5       56.5                0.92   
13                        53.8       71.2       71.2                0.91   
14                        71.2       73.6       73.6                0.96   
15                        93.4       97.5       97.5                0.94   
16                         NaN        NaN        NaN                 NaN   
17                         NaN        NaN        NaN                 NaN   
18                         NaN        NaN        NaN                 NaN   
19                         NaN        NaN        NaN                 NaN   
20                         NaN        NaN        NaN                 NaN   
21                         NaN        NaN        NaN                 NaN   
22                         NaN        NaN        NaN                 NaN   
23                         NaN        NaN        NaN                 NaN   
24                         NaN        NaN        NaN                 NaN   

   Unnamed: 6 Unnamed: 7  
0         NaN        NaN  
1         NaN        NaN  
2         NaN        NaN  
3         NaN        NaN  
4         NaN        NaN  
5       3-min    0.5-min  
6        0.92       0.83  
7        0.93   

# Interpretazione dei dati


### data1 - data2
Entrambi i dataset organizzati in un formato descrittivo, non adatto a processi di data analysis, cleaning o imputation.

Entrambi i dataset sono divisi in tabelle che contengono:
- Header (giorno, mese, anno, iterazione (run), intervallo orario)
- relhgt (altezza della fonte di SO2 dal terreno)

Colonne:
- Post: identificativo del sensore (intero)
- Degrees: angolo di inclinazione del sensore rispetto al sensore #1 (gradi)
- Azimuth: angolo di inclinazione del sensore rispetto al Nord (gradi)
- Wind-Dir: rappresenta quanto spesso la velocità del vento misurata è osservata al sensore. Questa misura è rappresentata in percentuale. La somma delle percentuali è posta in fondo alla rispettiva colonna (Non eguaglia sempre il 100%)
- Concentrazioni: colonna che racchiude 3 misure di concentrazione in $mg/m^3$ per le tre distanze considerate: 50, 100, 200 m

### Round Hill 2
Il secondo dataset è strutturato nello stesso modo. Le misure di concentrazione sono, questa volta, 9 in totale, 3 per ogni intervallo di tempo (10 min, 3 min, 1.5 min).<br>*La colonna relativa alla misurazione di concentrazione a 1.5 min è indicata con con "0.5-min", questo è probabilmente un errore avvenuto durante la trascrizione del dataset cartaceo*.<br>
Sono inoltre presenti misurazioni verticali, rispetto ai sensori, della concentrazione, rispettivamente a 2.5, 1.0, 0.5 m di altezza.



### meteo1 - meteo2
I due fogli contengono dati meteorologici registrati durante le misurazioni di concentrazione del gas.
Le colonne rappresentano:
- U (m/s) : velocità del vento orizzontale alla fonte
- Sa (deg) : deviazione standard dell'angolo di elevazione del vento misurato
- Se (deg) - Non presente in meteo2 : deviazione standard dell'angolo orizzontale del vento misurato
- U1.5, U3.0, U6.0, U12.0 (m/s) : velocità orizzontale del vento misurata ad altezze diverse (1.5 m, 3.0 m, 6.0 m, 12.0 m) da torri di osservazione situate nelle vicinanze della fonte del gas
- T1.5, T3.0, T6.0, T12.0 (C) : temperatura misurata dalle torri alle stesse altezze

Valori non rilevati vengono rappresentati da -99

### correction1 - correction2
I due fogli contengono i fattori di correzione moltiplicativi che tengono conto del tasso di evaporazione del gas. I valori sono compresi tra 0 e 1.0, riducendo la concentrazione effettiva del gas.
La colonna Q rappresenta il tasso di evaporazione del gas in g/s